ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -13,8 +13,59 @@
 
 package types
 
+import "fmt"
+
 const (
-	BlackHolePortFaultType = "network-blackhole-port"
-	LatencyFaultType       = "network-latency"
-	PacketLossFaultType    = "network-packet-loss"
+	BlackHolePortFaultType    = "network-blackhole-port"
+	LatencyFaultType          = "network-latency"
+	PacketLossFaultType       = "network-packet-loss"
+	missingRequiredFieldError = "required parameter %s is missing"
 )
+
+type NetworkFaultRequest interface {
+	ValidateRequest() error
+	ToString() (string, error)
+}
+
+type NetworkBlackholePortRequest struct {
+	Port        *int    `json:"Port"`
+	Protocol    *string `json:"Protocol"`
+	TrafficType *string `json:"TrafficType"`
+}
+
+type NetworkFaultInjectionResponse struct {
+	Status string `json:"status,omitempty"`
+	Error  string `json:"error,omitempty"`
+}
+
+func (request NetworkBlackholePortRequest) ValidateRequest() error {
+	if request.Port == nil {
+		return fmt.Errorf(missingRequiredFieldError, "Port")
+	}
+	if request.Protocol == nil || *request.Protocol == "" {
+		return fmt.Errorf(missingRequiredFieldError, "Protocol")
+	}
+	if request.TrafficType == nil || *request.TrafficType == "" {
+		return fmt.Errorf(missingRequiredFieldError, "TrafficType")
+	}
```

Comment:
Let's validate that `TrafficType` is either `ingress` or `egress` and nothing else.

Reply:
Was thinking this can be done in a future implementation...sure I will add this check